File: react-app/src/classes/HubEvents.ts
export class HubEvents {
  //Connection
  static OnReceiveConnectionId: string = "OnReceiveConnectionId";
  static SendHeartbeat: string = "SendHeartbeat";
  static OnSendHeartbeat: string = "OnSendHeartbeat";

  //Rooms
  static onRoomUpdated: string = "OnRoomUpdated";
  static onListOfRoomsUpdated: string = "OnListOfRoomsUpdated";

  //User
  static KickOut: string = "KickOut";
  static SetAdminStatus: string = "SetAdminStatus";
  static OnJoinRoom: string = "OnJoinRoom";
  static OnLeaveRoom: string = "OnLeaveRoom";
  static OnKickOut: string = "OnKickOut";
  static OnSetAdminStatus: string = "OnSetAdminStatus";

  //VideoPlayer
  static SetIsVideoPlaying: string = "SetIsVideoPlaying";
  static SetPlayedSeconds: string = "SetPlayedSeconds";
  static OnSetVideoUrl: string = "OnSetVideoUrl";
  static OnSetIsVideoPlaying: string = "OnSetIsVideoPlaying";
  static OnSetPlayedSeconds: string = "OnSetPlayedSeconds";

  //Chat
  static AddChatMessage: string = "AddChatMessage";
  static OnAddChatMessage: string = "OnAddChatMessage";

  //Playlist
  static AddPlaylistVideo: string = "AddPlaylistVideo";
  static DeletePlaylistVideo: string = "DeletePlaylistVideo";
  static OnAddPlaylistVideo: string = "OnAddPlaylistVideo";
  static OnDeletePlaylistVideo: string = "OnDeletePlaylistVideo";

  //RoomSettings
  static SetRoomPassword: string = "SetRoomPassword";
  static OnSetRoomPassword: string = "OnSetRoomPassword";

  //UserPermissions
  static SetUserPermissions: string = "SetUserPermissions";
  static OnSetUserPermissions: string = "OnSetUserPermissions";

}

File: dotnet-server/SignalR/HubEvents.cs
public static class HubEvents {
    //Connection
    public const string OnReceiveConnectionId = "OnReceiveConnectionId";

    //Room
    public const string KickOut = "KickOut";
    public const string SetAdminStatus = "SetAdminStatus";
    public const string OnJoinRoom = "OnJoinRoom";
    public const string OnLeaveRoom = "OnLeaveRoom";
    public const string OnKickOut = "OnKickOut";
    public const string OnSetAdminStatus = "OnSetAdminStatus";

    //VideoPlayer
    public const string SetIsVideoPlaying = "SetIsVideoPlaying";
    public const string SetPlayedSeconds = "SetPlayedSeconds";
    public const string OnSetVideoUrl = "OnSetVideoUrl";
    public const string OnSetIsVideoPlaying = "OnSetIsVideoPlaying";
    public const string OnSetPlayedSeconds = "OnSetPlayedSeconds";

    //Chat
    public const string AddChatMessage = "AddChatMessage";
    public const string OnAddChatMessage = "OnAddChatMessage";

    //Playlist
    public const string AddPlaylistVideo = "AddPlaylistVideo";
    public const string DeletePlaylistVideo = "DeletePlaylistVideo";
    public const string OnAddPlaylistVideo = "OnAddPlaylistVideo";
    public const string OnDeletePlaylistVideo = "OnDeletePlaylistVideo";

    //RoomSettings
    public const string SetRoomPassword = "SetRoomPassword";
    public const string OnSetRoomPassword = "OnSetRoomPassword";

    //UserPermissions
    public const string SetUserPermissions = "SetUserPermissions";
    public const string OnSetUserPermissions = "OnSetUserPermissions";
}